src/schema/local_discovery/city.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  GraphQLFloat,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+
+import Show from "schema/show"
+import PartnerShowSorts from "schema/sorts/partner_show_sorts"
+import Fair from "schema/fair"
+import FairSorts from "schema/sorts/fair_sorts"
+import EventStatus from "schema/input_fields/event_status"
+
+import cityData from "./city_data.json"
+
+const LOCAL_DISCOVERY_RADIUS_KM = 75
+
+const LatLngType = new GraphQLObjectType({
+  name: "LatLng",
+  fields: {
+    lat: {
+      type: GraphQLFloat,
+      resolve: obj => {
+        return obj.lat
+      },
+    },
+    lng: {
+      type: GraphQLFloat,
+      resolve: obj => {
+        return obj.lng
+      },
+    },
+  },
+})
+
+const LocalDiscoveryCityType = new GraphQLObjectType({
+  name: "LocalDiscoveryCity",
+  fields: {
+    slug: {
+      type: GraphQLString,
+      resolve: obj => {
+        return obj.slug
+      },
+    },
+    name: {
+      type: GraphQLString,
+      resolve: obj => {
+        return obj.name
+      },
+    },
+    coordinates: {
+      type: LatLngType,
+      resolve: obj => {
+        return obj.coordinates
+      },
+    },
+    shows: {
+      type: new GraphQLList(Show.type),
+      args: {
+        size: { type: GraphQLInt },
+        sort: PartnerShowSorts,
+        status: EventStatus,
+      },
+      resolve: (obj, args, _context, { rootValue: { showsLoader } }) => {
+        let gravityOptions = {
+          ...args,
+          displayable: true,
+          near: `${obj.coordinates.lat},${obj.coordinates.lng}`,
+          max_distance: LOCAL_DISCOVERY_RADIUS_KM,
+        }
+
+        // here is where we could fetch non-partner shows and
+        // merge them in with partner shows.
+        //
+        // but how to deal with sorting across two disparate
+        // collections?
```

Comment:
I know you know that we discussed this on the notion doc, but leaving here for posterity that we shouldn’t be sorting large paginated lists on MP, as it would require loading all data.

Reply:
Agreed, that should most likely happen in Gravity if that's where we end up sorting Non Partner models (sql join?). It would be way faster too

Reply:
Yeah, looks like that's where we're ending up anyway, after this morning's platform discussion. Hopefully that smooths the way forward, at least from MP's perspective.